src/query/storages/parquet/src/parquet2/pruning.rs

```diff
@@ -261,14 +262,18 @@ impl PartitionPruner {
             let (sub_stats, parts) =
                 self.read_and_prune_file_meta(path, file_meta, operator.clone())?;
 
-            for p in parts {
+            let mut groups = HashMap::with_capacity(parts.len());
+
+            for (gid, p) in parts.into_iter().enumerate() {
                 max_compression_ratio = max_compression_ratio
                     .max(p.uncompressed_size() as f64 / p.compressed_size() as f64);
                 max_compressed_size = max_compressed_size.max(p.compressed_size());
-                partitions.push(Arc::new(
-                    Box::new(ParquetPart::Parquet2RowGroup(p)) as Box<dyn PartInfo>
-                ));
+                groups.insert(gid, p);
```

Comment:
sorry for the late review. 
do you mean each file as a part? not make sense to me.

Reply:
maybe u can expain amore in comments and/or the pr summary. so we can understand your code better.

Reply:
i am fixing it , the code would be push tonight.

Reply:
@youngsofun  please help check